### PR TITLE
Modifying home controller

### PR DESCRIPTION
**Pull request by ula0156**
1. Created a filter class in order to filter items based on the specified criterions. 
2. Modify productPickes by adding a filter as a parameter. 
3. Edited HomeController index method.

### CODE REVIEW

File: onlineShop.core/ProductPicker/Filters.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                if (!t.IsInstanceOfType(product) || !InStock(product, stock))
+                {
+                    return false;
+                }
+                
+                return true;
+            });
+        }
+        public static bool InStock(Product product, int stock, Type t)
```
Comment (simionliviumihai) [inline onlineShop.core/ProductPicker/Filters.cs:26]: add empty line

File: onlineShop.core/ProductPicker/IProductPicker.cs [hunk 2]
```diff
@@ -4,8 +4,12 @@
 
 namespace onlineShop.ProductPickers
 {
+    // what is stock??
```
Comment (simionliviumihai) [inline onlineShop.core/ProductPicker/IProductPicker.cs:7]: remove

File: onlineShop.core/Products/Book.cs [hunk 3]
```diff
@@ -8,14 +8,18 @@ public class Book: PhysicalProduct
 
         public string Author { get; set; }
 
+        public string Genre { get; set; }
```
Comment (simionliviumihai) [inline onlineShop.core/Products/Book.cs:11]: better use tags to put the genre. A book can have multiple genres. e.g. scifi, romance

Comment (simionliviumihai) [reply onlineShop.core/Products/Book.cs:11]: And get rid of this genre attribute

File: onlineShopWeb/DataAccess/ProvidersFactory.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using onlineShop.Data;
+using onlineShop.Data.Database;
+using System;
+using System.Collections.Generic;
```
Comment (simionliviumihai) [inline onlineShopWeb/DataAccess/ProvidersFactory.cs:4]: I think some of these using are not necessary.

**Review by simionliviumihai — APPROVED** [review]

File: onlineShop.core/ProductPicker/Filters.cs (hunk 1 above)
Comment (ula0156) [reply onlineShop.core/ProductPicker/Filters.cs:26]: done

File: onlineShop.core/ProductPicker/IProductPicker.cs (hunk 2 above)
Comment (ula0156) [reply onlineShop.core/ProductPicker/IProductPicker.cs:7]: done

File: onlineShop.core/Products/Book.cs (hunk 3 above)
Comment (ula0156) [reply onlineShop.core/Products/Book.cs:11]: Disagree.  
I'd like to keep genre.

File: onlineShopWeb/DataAccess/ProvidersFactory.cs (hunk 4 above)
Comment (ula0156) [reply onlineShopWeb/DataAccess/ProvidersFactory.cs:4]: done

**Pull request merged**